multiset/multiset.pyi

```diff
@@ -84,7 +86,14 @@ class BaseMultiset(Mapping[TElement, int], Generic[TElement]):
 class Multiset(BaseMultiset[TElement], MutableMapping[TElement, int], Generic[TElement]):
     def __setitem__(self, element: TElement, multiplicity: int) -> None: ...
     def __delitem__(self, element: TElement) -> None: ...
+    @overload
     def update(self, *others: OtherType) -> None: ...
+    @overload
+    def update(self, __m: SupportsKeysAndGetItem[T, int], /, **kwargs: int) -> None: ...
```

Comment:
I'm not too sure about these changes, I just did what `typing.reveal_type` told me would work and then mypy stopped complaining. The SupportsKeysAndGetItem thing means you can do something like this:

```
>>> import multiset
>>> x = multiset.Multiset()
>>> y = {'a': 1, 'b': 2}
>>> x.update(y)
>>> x
Multiset({'a': 1, 'b': 2})
```

but the `**kwargs` part confuses me. Neither set nor Multiset seem to allow kwargs in a call to update. I'll try removing that …

Reply:
With the old stubs, stubtest says:

```
multiset/setup.py:2: error: Skipping analyzing "setuptools": module is installed, but missing library stubs or py.typed marker  [import-untyped]
multiset/multiset/multiset.pyi:87: error: Signature of "update" incompatible with supertype "MutableMapping"  [override]
multiset/multiset/multiset.pyi:87: note:      Superclass:
multiset/multiset/multiset.pyi:87: note:          @overload
multiset/multiset/multiset.pyi:87: note:          def update(self, SupportsKeysAndGetItem[TElement, int], /, **kwargs: int) -> None
multiset/multiset/multiset.pyi:87: note:          @overload
multiset/multiset/multiset.pyi:87: note:          def update(self, Iterable[tuple[TElement, int]], /, **kwargs: int) -> None
multiset/multiset/multiset.pyi:87: note:          @overload
multiset/multiset/multiset.pyi:87: note:          def update(self, **kwargs: int) -> None
multiset/multiset/multiset.pyi:87: note:      Subclass:
multiset/multiset/multiset.pyi:87: note:          def update(self, *others: Iterable[Any] | Mapping[Any, int]) -> None
```

which is where I got it from.

Reply:
I'm not really sure why it thinks the superclass has this signature because the inheritance is object -> BaseMultiset -> Multiset.

Reply:
Oh, I see, it's because of the `MutableMapping[TElement, int]` in the class declaration in the stubs.

Reply:
So perhaps a better change is to remove that from the class declaration in the stubs? Then no other changes to the stubs are required. I'm not sure what's that meant to represent and if that will make mypy tell folks that something that should be possible is wrong.

Reply:
Apparently also dict.update supports the kwargs. So technically Multiset should was well, because MutableMap requires it (which I think it did not back when I first implemented it). So I think this is fine, but I should also update the implementation to match...